### ISSUE TITLE
Not Working on MX Linux 

### ISSUE THREAD

**Issue by biterminator** [Windscribe/Desktop-App#14]
Windscribe GUI  is not working on MX Linux 21, debian based Linux OS, that uses sysvinit instead of systemd.
https://mxlinux.org/

_Windscribe helper initialize error. Please reinstall the application or contact support._

**Comment by KITKATKILLER67** [comment]
Probably an issue with digital signatures.

**Comment by bernerdad** [comment]
Hi.  We'll do some testing on distros using sysvinit asap.

**Comment by alok-sin** [comment]
Issue is because of `systemctl` commands that seem to start the deamon

**Comment by koplenov** [comment]
Are there any changes?

It still doesn't work

**Comment by bernerdad** [comment]
Have your tried v2.4.8 from our Guinea Pig update channel?

**Comment by koplenov** [comment]
Where i can change update channel?

// App not start

![image](https://user-images.githubusercontent.com/25139987/171827600-62c2e253-a358-48df-9a10-c5d823fea223.png)

**Comment by bernerdad** [comment]
Give this version a shot.  It is our current beta release.
[windscribe_2.4.10_amd64.deb.zip](https://github.com/Windscribe/Desktop-App/files/8833312/windscribe_2.4.10_amd64.deb.zip)

**Comment by koplenov** [comment]
Hmm, same error
![image](https://user-images.githubusercontent.com/25139987/171909602-5ae48700-90e9-4bfe-aeb8-85fb9d0df76b.png)

**Comment by bernerdad** [comment]
Drat!  Is there anything in `/usr/local/windscribe/helper_log.txt` indicating why the helper failed?

**Comment by koplenov** [comment]
Nothing useful. Where else could the logs be?
![image](https://user-images.githubusercontent.com/25139987/172131870-9e364594-1d49-45cb-8e00-e1526c7afcfd.png)

**Comment by koplenov** [comment]
I think I acquired problems during the first installation of windscribe.

I'll try to fix it and try again

![image](https://user-images.githubusercontent.com/25139987/172174564-a18e28c5-648c-476f-a45b-a038dd087465.png)

**Comment by koplenov** [comment]
Well.

I fixed the package manager which broke after installing the regular version of windscribe.

I uninstalled the old version and everything related to it.


Installed the version from our Guinea Pig update channel.

Installed WITHOUT ERRORS this time.

But, it doesn't work.

No logs showed up either.

How can I provide more information about the problem?

**Comment by koplenov** [comment]
Cool, we're waiting.

**Comment by alok-sin** [comment]
> We've created a ticket to investigate this issue. However, MX Linux is not one of our officially supported distributions, so it may take a while before one of our developers can look into this.

Alpine and TinyCore are also affected in the same way as far as I can tell. They are the go-to repos for containerization and we would appreciate it a lot if you can take a look at it. 
Meanwhile, let us know if we can help with anything. Cheers :)

**Comment by Ivantheallknowing** [comment]
You can try booting your system with systemd and the helper should work after that

![image](https://user-images.githubusercontent.com/115170056/194346834-8bbc6049-f751-4740-b84d-15f29241d10d.png)

**Comment by hopeless-attempt** [comment]
Got the current release (2.5.17)  working on MXLinux running SysVInit by the following.

Completely uninstall Windscribe.
Boot MX with systemd.
Install Windscribe via the deb file. Installed with no issues.
Boot MX with SysVInit.
Create a script to run /usr/local/windscribe/helper as sudo at login.
Then no problems running Windscribe under SysVInit.

The Windscribe GUI obviously needs the helper program running to work.
Also the update from the GUI won't work under SysVInit so I have to reboot to systemd to do that.

It's a pain but at least this work-around let's me use Windscribe on SysVInit.
Noticed that the MX Package Installer has now dropped Windscribe . No wonder.

**Comment by sith-on-mars** [comment]
> Got the current release (2.5.17) working on MXLinux running SysVInit by the following.
> 
> Completely uninstall Windscribe. Boot MX with systemd. Install Windscribe via the deb file. Installed with no issues. Boot MX with SysVInit. Create a script to run /usr/local/windscribe/helper as sudo at login. Then no problems running Windscribe under SysVInit.
> 
> The Windscribe GUI obviously needs the helper program running to work. Also the update from the GUI won't work under SysVInit so I have to reboot to systemd to do that.
> 
> It's a pain but at least this work-around let's me use Windscribe on SysVInit. Noticed that the MX Package Installer has now dropped Windscribe . No wonder.

I'm on Guinea Pig update channel, and also had problem with launching helper on OpenSUSE MicroOS (an immutable system) . The workaround above works.

Here is the desktop file I use:

```
[Desktop Entry]
Name=Windscribe Helper
Comment=Windscribe start-up script
Exec=pkexec /opt/windscribe/helper
Terminal=false
Type=Application
```

**Comment by hopeless-attempt** [comment]
To sith-on-mars,

Glad you got it working. I don't know anything about OpenSUSE Micros or even what an immutable system is and I know the following is not good OPSEC but if you want a lazy way to launch the helper program as sudo without having to enter your admin password the following works.

#!/bin/bash
#
adminpassword="XXXXXXXXXX"
printf "$adminpassword""\n" | sudo -S /opt/windscribe/helper
#

If that is where your helper program is located. On my system it's at /usr/local/windscribe/helper.

MXLinux has now got Windscribe working properly with startup scripts in both systemd & SysVInit so I don't need to run my own script anymore.

**Comment by sith-on-mars** [comment]
@bernerdad  

I found on OpenSUSE MicroOS, the helper not launching issue was caused by the lack of Windscribe helper systemd service on boot. Windscribe created `windscribe-helper.service` in `/etc/systemd/system`, but didn't manage to enable systemd service. So the simple fix is to enable systemd service:

`sudo systemctl enable windscribe-helper.service`

So maybe Windscribe should make a launch script to enable systemd service?

This is tested on the lastest Guinea Pig update channel. I hope it helps!
